op-program/host/host.go

```diff
@@ -85,8 +66,14 @@ func FaultProofProgram(logger log.Logger, cfg *config.Config) error {
 		return fmt.Errorf("failed to create hints pipe: %w", err)
 	}
 	defer hHostRW.Close()
-	hHost := preimage.NewHintReader(hHostRW)
-	routeHints(logger, hHost, hinter)
+
+	go func() {
+		err := PreimageServer(ctx, logger, cfg, pHostRW, hHostRW)
+		if err != nil {
+			logger.Error("preimage server failed", "err", err)
+		}
+		logger.Debug("Preimage server stopped")
```

Comment:
move L68 (`defer hHostRW.Close()`)  here so that we close the socket before blocking on child wait. Doing this should also fully resolve [CLI-3868](https://linear.app/optimism/issue/CLI-3868/exit-when-pre-image-data-missing).

Reply:
Done. I was wondering about that but got tied up in various questions about what the right approach to ensuring things closed and when etc so had left it.